doc/cephadm/services/smb.rst

```diff
@@ -0,0 +1,205 @@
+.. _deploy-cephadm-smb-samba:
+
+===========
+SMB Service
+===========
+
+.. note:: Only the SMB3 protocol is supported.
+
+.. warning::
+
+    SMB support is under active development and many features may be
+    missing or immature. Additionally, a Manager module to automate
+    smb clusters and smb shares is in development. Once that feature
+    is developed it will be the preferred method for managing
+    smb on ceph.
+
+
+Deploying Samba Containers
+==========================
+
+Cephadm deploys `Samba <http://www.samba.org>`_ servers using container images
+built by the `samba-container project <http://github.com/samba-in-kubernetes/samba-container>`_.
+
+In order to host SMB Shares with access to CephFS file systems, deploy
+Samba Containers with the following command:
+
+.. prompt:: bash #
+
+    orch apply smb <cluster_id> <config_uri> [--features ...] [--placement ...] ...
+
+There are a number of additional parameters that the command accepts. See
+the Service Specification for a description of these options.
+
+Service Specification
+=====================
+
+An SMB Service can be applied using a specification. An example in YAML follows:
+
+.. code-block:: yaml
+
+    service_type: smb
+    service_id: tango
+    placement:
+      hosts:
+        - ceph0
+    spec:
+      cluster_id: tango
+      features:
+        - domain
+      config_uri: rados://.smb/tango/scc.toml
+      custom_dns:
+        - "192.168.76.204"
+      join_sources:
+        - "rados:mon-config-key:smb/config/tango/join1.json"
+      include_ceph_users:
+        - client.smb.fs.cluster.tango
+
+The specification can then be applied by running the following command:
+
+.. prompt:: bash #
+
+   ceph orch apply -i smb.yaml
+
+
+Service Spec Options
+--------------------
+
+Fields specific to the ``spec`` section of the SMB Service are described below.
+
+cluster_id
+    A short name identifying the smb "cluster". In this case a cluster is
+    simply a management unit of one or more Samba services sharing a common
+    configuration, and may not provide actual clustering or availability
+    mechanisms.
+
+features
+    A list of pre-defined terms enabling specific deployment characteristics.
+    An empty list is valid. Supported terms:
+
+    * ``domain``: Enable domain member mode
+
+config_uri
+    A string containing a (standard or de-facto) URI that identifies a
+    configuration source that should be loaded by the samba-container as the
+    primary configuration file.
+    Supported URI schemes include ``http:``, ``https:``, ``rados:``, and
```

Comment:
Will this include the ability to specify the SMB protocol version?

Reply:
This config uri (and join sources) tell the samba container where to pull configuration properties from. It doesn't effect the what SMB protcol directly. One could add a custom config file enabling SMBv1. However, our plan is to only support SMBv2 & SMBv3 in the higher level management when it gets added.